imap-proto/src/parser/rfc3501.rs

```diff
@@ -169,16 +157,48 @@ named!(resp_text_code<ResponseCode>, do_parse!(
     (coded)
 ));
 
-named!(capability<&str>, do_parse!(
-    tag_s!(" ") >>
-    atom: map_res!(take_till1_s!(atom_specials), str::from_utf8) >>
-    (atom)
+/// [RFC3501 - 9. Formal Syntax - capability](https://tools.ietf.org/html/rfc3501#section-9)
+named!(capability_auth<Capability>, do_parse!(
+    tag_s!("AUTH=") >>
+    auth: atom >>
+    (Capability::Auth(auth))
+));
+
+/// [RFC3501 - 9. Formal Syntax - capability](https://tools.ietf.org/html/rfc3501#section-9)
+named!(capability_atom<Capability>, do_parse!(
```

Comment:
On second thought, both this and `capability_auth` could just be map!s in capability, there isn't really any need for them to be separate